site/_posts/2023-02-03-single-sign-on-laravel-fusionauth.md

```diff
@@ -0,0 +1,388 @@
+            $table->dropColumn('fusionauth_refresh_token');
+            $table->string('password')->nullable(false)->change();
+        });
+    }
+};
+```
+
+We also need to add `doctrine/dbal` as a dependency, which is a package responsible for changing table and column
+definitions:
+
+```shell
+$ ./vendor/bin/sail composer require doctrine/dbal
+```
+
+Let's run our migration to add those new columns:
+
+```shell
+$ ./vendor/bin/sail php artisan migrate
+
+   INFO  Running migrations.  
+
+  2023_02_03_123000_add_fusionauth_fields_user_table ........................... 51ms DONE
+```
+
+Now, let's change our User model so it can become aware of these new columns. Modify `app/Models/User.php` and add them
+to both `$fillable` and `$hidden` properties:
+
+```php
+/**
+ * The attributes that are mass assignable.
+ *
+ * @var array<int, string>
+ */
+protected $fillable = [
+    'name',
+    'email',
+    'password',
+    'fusionauth_id',            // Add this column here
+    'fusionauth_token',         // Add this column here
+    'fusionauth_refresh_token', // Add this column here
+];
+
+/**
+ * The attributes that should be hidden for serialization.
+ *
+ * @var array<int, string>
+ */
+protected $hidden = [
+    'password',
+    'remember_token',
+    'fusionauth_id',            // Add this column here
+    'fusionauth_token',         // Add this column here
+    'fusionauth_refresh_token', // Add this column here
+];
+```
+
+#### Adding routes
+
+Now that our `Users` model and table have more details about the integration, let's define the necessary routes to
+redirect the user to the FusionAuth login and the page that the user will be redirected to after completing the process
+there. We need to add those routes to our `routes/web.php` file:
+
+```php
+use App\Models\User;
+use Illuminate\Support\Facades\Auth;
+use Laravel\Socialite\Facades\Socialite;
+
+/**
+ * Responsible for redirecting the user to the FusionAuth login page 
+ */
+Route::get('/auth/redirect', function () {
+    return Socialite::driver('fusionauth')->redirect();
+})->name('login');
+
+/**
+ * This is the address that we configured in our .env file which the user will be redirected to after completing the
+ * login process
+ */
+Route::get('/auth/callback', function () {
+    /** @var \SocialiteProviders\Manager\OAuth2\User $user */
+    $user = Socialite::driver('fusionauth')->user();
+
+    // Let's create a new entry in our users table (or update if it already exists) with some information from the user  
+    $user = User::updateOrCreate([
+        'fusionauth_id' => $user->id,
+    ], [
+        'name' => $user->name,
+        'email' => $user->email,
+        'fusionauth_token' => $user->token,
+        'fusionauth_refresh_token' => $user->refreshToken,
+    ]);
+
+    // Logging the user in
+    Auth::login($user);
+
+    // Here, you should redirect to your app's authenticated pages (e.g. the user dashboard)
+    return redirect('/');
```

Comment:
How do these pages check that the user is authenticated? is it the `@auth` markup in the view?

Reply:
Yes, this is a [Blade directive](https://laravel.com/docs/9.x/blade#authentication-directives)